### PR TITLE
AA: Implement KSErrorType

### PR DESCRIPTION
**Pull request by neetopia**
also includes an implementation for KSTypeAlias, as well as some refactoring for common logics for KSDeclaration.

### CODE REVIEW

File: kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/KSValueArgumentImpl.kt [hunk 1]
```diff
@@ -51,4 +51,8 @@ class KSValueArgumentImpl private constructor(
     override fun <D, R> accept(visitor: KSVisitor<D, R>, data: D): R {
         return visitor.visitValueArgument(this, data)
     }
+
+    override fun toString(): String {
+        return "${name?.asString() ?: ""}:$value"
```
Comment (jsjeon) [inline kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/KSValueArgumentImpl.kt:56]: If `name` is `null`, we'll get something like `:v`?

Also, if it is a named argument, I expected "name = value" instead.

File: kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/KSErrorType.kt [hunk 2]
```diff
@@ -0,0 +1,43 @@
+package com.google.devtools.ksp.impl.symbol.kotlin
```
Comment (jsjeon) [inline kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/KSErrorType.kt:1]: Copyright

File: kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/KSErrorTypeClassDeclaration.kt [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    override val simpleName: KSName = KSNameImpl.getCached("<Error>")
+
+    override val superTypes: Sequence<KSTypeReference> = emptySequence()
+
+    override val typeParameters: List<KSTypeParameter> = emptyList()
+
+    override fun getSealedSubclasses(): Sequence<KSClassDeclaration> = emptySequence()
+
+    override fun asStarProjectedType(): KSType {
+        TODO()
```
Comment (jsjeon) [inline kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/KSErrorTypeClassDeclaration.kt:62]: Nit: seems safe to use `error(...)`, like a message with "Should not be called"? Below too.

File: kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/KSTypeImpl.kt [hunk 4]
```diff
@@ -29,18 +29,22 @@ import org.jetbrains.kotlin.analysis.api.types.KtFunctionalType
 import org.jetbrains.kotlin.analysis.api.types.KtNonErrorClassType
 import org.jetbrains.kotlin.analysis.api.types.KtType
 import org.jetbrains.kotlin.analysis.api.types.KtTypeNullability
+import org.jetbrains.kotlin.utils.addToStdlib.safeAs
 
-class KSTypeImpl private constructor(private val type: KtType) : KSType {
+class KSTypeImpl private constructor(val type: KtType) : KSType {
```
Comment (jsjeon) [inline kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/KSTypeImpl.kt:34]: Isn't `internal` good enough? I'm afraid it's okay to expose element from Analysis API.

File: kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/KSValueArgumentImpl.kt (hunk 1 above)
Comment (neetopia) [reply kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/KSValueArgumentImpl.kt:56]: yes we will get something like you said. the `toString()` method currently tries to align with the compiler plugin implementation, might change it later if this is better representation.

**Review by jsjeon — APPROVED** [review]

**Pull request merged**